src/OrchardCore.Modules/OrchardCore.Workflows/Views/Activity.Design.cshtml

```diff
@@ -10,8 +10,10 @@
     var hasEditor = activity.HasEditor;
     var activityType = isEvent ? "Event" : "Task";
     var activityTypeClass = isEvent ? "fa-bolt" : "fa-caret-right";
+    var xPosition = activityRecord.X <= 0 ? 0 : activityRecord.X;
```

Comment:
Not sure if this is the right place to do this. We could potentially add a check when saving the activityRecord or on the getter / setter of the X, Y properties on the record.

Reply:
I would do it here as well, since it seems to be the safest place. If you would do it when saving, and if for some reason the X Y coordinates gets changed to negative values due to some bug, we'd still have the issue you are fixing.